rgw/rgw.go

```diff
@@ -0,0 +1,805 @@
+package rgw
+
+/*
+#cgo LDFLAGS: -lrgw
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <rados/librgw.h>
+#include <rados/rgw_file.h>
+
+// readdir_callback.go
+extern bool common_readdir_cb(const char *name, void *arg, uint64_t offset,
+                       struct stat *st, uint32_t mask,
+                       uint32_t flags);
+*/
+import "C"
+
+import (
+	"syscall"
+	"unsafe"
+
+	gopointer "github.com/mattn/go-pointer"
+)
+
+// RGW represents a librgw handle
+//
+// typedef void* librgw_t;
+type RGW struct {
+	libRGW *C.librgw_t
+}
+
+// Implements:
+//   int librgw_create(librgw_t *rgw, int argc, char **argv)
+func createRGW(argc C.int, argv **C.char) (*RGW, error) {
+	rgw := &RGW{}
+	var libRGW C.librgw_t
+
+	ret := C.librgw_create(&libRGW, argc, argv)
+	if ret == 0 {
+		rgw.libRGW = &libRGW
+		return rgw, nil
+	}
+	return nil, getError(ret)
+
+}
+
+// CreateRGW create a RGW instance
+func CreateRGW(argv []string) (*RGW, error) {
+	cargv := make([]*C.char, len(argv))
+	for i := range argv {
+		cargv[i] = C.CString(argv[i])
+		defer C.free(unsafe.Pointer(cargv[i]))
+	}
+
+	return createRGW(C.int(len(cargv)), &cargv[0])
+}
+
+// ShutdownRGW shutdown a RGW instance
+//
+// Implements:
+//   void librgw_shutdown(librgw_t rgw)
+func ShutdownRGW(rgw *RGW) {
+	C.librgw_shutdown(*rgw.libRGW)
+}
+
+// FS represents a rgw fs
+//
+// FS exports ceph's rgw_fs from include/rados/rgw_file.h
+type FS struct {
+	rgwFS *C.struct_rgw_fs
+}
+
+// MountFlag is used to control behavior of Mount()
+type MountFlag uint32
+
+const (
+	// MountFlagNone keep the default behavior of Mount()
```

Comment:
The word "keep" here is a bit confusing. Perhaps "MountFlagNone selects the default behavior of Mount" would be better?